src/eve/type_definitions.py

```diff
@@ -56,41 +56,43 @@ class Enum(enum.Enum):
 
     @classmethod
     def __get_validators__(cls) -> PydanticCallableGenerator:
-        yield cls._strict_type_validator
+        yield cls._value_validator
 
     @classmethod
-    def _strict_type_validator(cls, v: Any) -> Enum:
-        if not isinstance(v, cls):
-            raise TypeError(f"Invalid value type [expected: {cls}, received: {v.__class__}]")
-        return v
+    def _value_validator(cls, v: Any) -> Enum:
+        if isinstance(v, cls):
+            return v
+        return cls(v)  # throws a ValueError if value is not contained in enum
 
 
 class IntEnum(enum.IntEnum):
     """Basic :class:`enum.IntEnum` subclass with strict type validation."""
 
     @classmethod
     def __get_validators__(cls) -> PydanticCallableGenerator:
-        yield cls._strict_type_validator
+        yield cls._value_validator
 
     @classmethod
-    def _strict_type_validator(cls, v: Any) -> IntEnum:
-        if not isinstance(v, cls):
-            raise TypeError(f"Invalid value type [expected: {cls}, received: {v.__class__}]")
-        return v
+    def _value_validator(cls, v: Any) -> IntEnum:
+        if isinstance(v, cls):
+            return v
+        return cls(v)  # throws a ValueError if value is not contained in enum
 
 
 class StrEnum(str, enum.Enum):
     """:class:`enum.Enum` subclass with strict type validation and supporting string operations."""
 
     @classmethod
     def __get_validators__(cls) -> PydanticCallableGenerator:
-        yield cls._strict_type_validator
+        yield cls._value_validator
 
     @classmethod
-    def _strict_type_validator(cls, v: Any) -> StrEnum:
-        if not isinstance(v, cls):
-            raise TypeError(f"Invalid value type [expected: {cls}, received: {v.__class__}]")
-        return v
+    def _value_validator(cls, v: Any) -> StrEnum:
+        if hasattr(cls, v):
+            v = cls[v]
```

Comment:
Either `v` is guaranteed to be a `str` and thus the type annotation is wrong or the call to `hasattr` can trigger an Exception if something else than a string is passed. I think the safe way to check if the passed value is a string with the name of an enum member is using the `__members__` dictionary of the enum class:

```suggestion
    def _value_validator(cls, v: Any) -> StrEnum:
        if isinstance(v, str) and v in cls.__members__:
            v = cls[v]
```

It's probably worth to add a comment here mentioning that the passed string is first looked up as a enum member and then as a enum value.